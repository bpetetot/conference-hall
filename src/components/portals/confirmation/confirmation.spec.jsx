/* eslint-env jest */
import React from 'react'
import snap from 'tests/snapshot'
import { mount } from 'enzyme'
import Button from 'components/button'
import ConfirmationPopin from './index.js'

const snapshot = props => snap(ConfirmationPopin)({ ...props })

describe('components/confirmationPopin', () => {
  it('should render', snapshot({}))
  it('should call onClick by default', () => {
    const wrapper = mount(<ConfirmationPopin
      title="Confirmation title"
      content="Can you confirm by clicking on OK, else click Cancel ?"
      withOk
      withCancel
      className="default-theme"
      renderTrigger={({ show }) => (
        <Button type="button" onClick={show}>
          Open Confirmation Popin
        </Button>
      )}
    />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.confirmation-actions').length).toEqual(0)
    wrapper.find('Button').simulate('click')
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('.confirmation-actions').length).toEqual(1)
  })
})
