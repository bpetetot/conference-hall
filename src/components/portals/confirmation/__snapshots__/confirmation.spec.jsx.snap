// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/confirmationPopin should call onClick by default 1`] = `
<ConfirmationPopin
  className="default-theme"
  content="Can you confirm by clicking on OK, else click Cancel ?"
  defaultOpen={false}
  renderTrigger={[Function]}
  title="Confirmation title"
  withCancel={true}
  withOk={true}
>
  <Modal
    className="confirmation-popin default-theme"
    defaultOpen={false}
    onClose={[Function]}
    renderTrigger={[Function]}
    withClickOutside={false}
    withCloseIcon={false}
    withEscapeClose={false}
  >
    <OpenTrigger
      defaultOpen={false}
      onClose={[Function]}
      renderTrigger={[Function]}
      withEscapeClose={false}
    >
      <Button
        accent={false}
        block={false}
        error={false}
        loading={false}
        onClick={[Function]}
        primary={true}
        secondary={false}
        simple={false}
        size="normal"
        tertiary={false}
        type="button"
      >
        <button
          className="cc-button cc-button-primary"
          onClick={[Function]}
          type="button"
        >
          Open Confirmation Popin
        </button>
      </Button>
    </OpenTrigger>
  </Modal>
</ConfirmationPopin>
`;

exports[`components/confirmationPopin should call onClick by default 2`] = `
<ConfirmationPopin
  className="default-theme"
  content="Can you confirm by clicking on OK, else click Cancel ?"
  defaultOpen={false}
  renderTrigger={[Function]}
  title="Confirmation title"
  withCancel={true}
  withOk={true}
>
  <Modal
    className="confirmation-popin default-theme"
    defaultOpen={false}
    onClose={[Function]}
    renderTrigger={[Function]}
    withClickOutside={false}
    withCloseIcon={false}
    withEscapeClose={false}
  >
    <OpenTrigger
      defaultOpen={false}
      onClose={[Function]}
      renderTrigger={[Function]}
      withEscapeClose={false}
    >
      <Button
        accent={false}
        block={false}
        error={false}
        loading={false}
        onClick={[Function]}
        primary={true}
        secondary={false}
        simple={false}
        size="normal"
        tertiary={false}
        type="button"
      >
        <button
          className="cc-button cc-button-primary"
          onClick={[Function]}
          type="button"
        >
          Open Confirmation Popin
        </button>
      </Button>
      <Portal>
        <Portal
          containerInfo={
            <div
              class="portal"
            >
              <div
                class="backdrop"
                role="button"
              />
              <div
                class="modal confirmation-popin default-theme"
              >
                <div
                  class="confirmation-text"
                >
                  <div
                    class="confirmation-title"
                  >
                    Confirmation title
                  </div>
                  <div
                    class="confirmation-content"
                  >
                    Can you confirm by clicking on OK, else click Cancel ?
                  </div>
                </div>
                <div
                  class="confirmation-actions"
                >
                  <button
                    class="cc-button cc-button-secondary"
                    type="button"
                  >
                    Cancel
                  </button>
                  <button
                    class="cc-button cc-button-accent cc-button-primary"
                    type="button"
                  >
                    Ok
                  </button>
                </div>
              </div>
            </div>
          }
        >
          <Backdrop
            onClick={[Function]}
            withClickOutside={false}
          >
            <div
              className="backdrop"
              onClick={[Function]}
              role="button"
            />
          </Backdrop>
          <div
            className="modal confirmation-popin default-theme"
          >
            <div
              className="confirmation-text"
            >
              <div
                className="confirmation-title"
              >
                Confirmation title
              </div>
              <div
                className="confirmation-content"
              >
                Can you confirm by clicking on OK, else click Cancel ?
              </div>
            </div>
            <div
              className="confirmation-actions"
            >
              <Button
                accent={false}
                block={false}
                error={false}
                loading={false}
                onClick={[Function]}
                primary={true}
                secondary={true}
                simple={false}
                size="normal"
                tertiary={false}
              >
                <button
                  className="cc-button cc-button-secondary"
                  onClick={[Function]}
                  type="button"
                >
                  Cancel
                </button>
              </Button>
              <Button
                accent={true}
                block={false}
                error={false}
                loading={false}
                onClick={[Function]}
                primary={true}
                secondary={false}
                simple={false}
                size="normal"
                tertiary={false}
              >
                <button
                  className="cc-button cc-button-accent cc-button-primary"
                  onClick={[Function]}
                  type="button"
                >
                  Ok
                </button>
              </Button>
            </div>
          </div>
        </Portal>
      </Portal>
    </OpenTrigger>
  </Modal>
</ConfirmationPopin>
`;

exports[`components/confirmationPopin should render 1`] = `null`;
